pkgs/development/compilers/dotnet/buildDotnet.nix

```diff
@@ -11,12 +11,17 @@ assert builtins.elem type [ "aspnetcore" "netcore" "sdk"];
 , libuuid
 , zlib
 , curl
-}: 
+}:
 let pname = if type == "aspnetcore" then "aspnetcore-runtime" else if type == "netcore" then "dotnet-runtime" else "dotnet-sdk";
+    platformNames = {
+      "x86_64-darwin" = "osx";
+      "x86_64-linux" = "linux";
+    };
+    platform = builtins.getAttr stdenv.system platformNames;
```

Comment:
this gets evaluated before meta.platforms gets checked.

Reply:
As an alternative, would it make sense to take `platform` as an argument here, meaning that the invocation over in `pkgs/development/compilers/dotnet/default.nix` includes the platform and platform-specific SHA together? I found it a little confusing that the invocation passes a SHA based on `if stdenv.isLinux` and relies on this function to identify the platform separately.